dts/bindings/display/galaxycore,gc9x01x.yaml

```diff
@@ -0,0 +1,169 @@
+# Copyright (c) 2023 Mr Beam Lasers GmbH.
+# Copyright (c) 2023 Amrith Venkat Kesavamoorthi <amrith@mr-beam.org>
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+  GC9X01X display controller.
+
+  Here is an example to define a display interface:
+
+  / {
+    pwmleds {
+      compatible = "pwm-leds";
+      backlight_pwm: backlight_pwm {
+        pwms = <&ledc0 0 PWM_MSEC(10) PWM_POLARITY_NORMAL>;
+      };
+    };
+  };
+
+  &spi2 {
+    gc9a01a_lcd: gc9a01a_lcd@0 {
+      compatible = "galaxycore,gc9x01x";
+      reg = <0>;
+      spi-max-frequency = <100000000>;
+      cmd-data-gpios = <&gpio0 4 GPIO_ACTIVE_HIGH>;
+      reset-gpios = <&gpio0 8 GPIO_ACTIVE_LOW>;
+      backlight-pwm = <&backlight_pwm>;
+
+      width = <240>;
+      height = <240>;
+      pixel-format = <GC9X01X_PIXEL_FORMAT_RGB565>;
+    };
+  };
+
+
+compatible: "galaxycore,gc9x01x"
+
+include: [spi-device.yaml, display-controller.yaml]
+
+properties:
+  reset-gpios:
+    type: phandle-array
+    required: true
+    description: RESET pin
+
+      The RESET pin of the GC9X01X is active low.
+      If connected directly the MCU pin should be configured
+      as active low.
+
+  cmd-data-gpios:
+    type: phandle-array
+    required: true
+    description: Command/Data pin
+      The Data/Command pin of the GC9X01X is to be configured
+      high(1) for data, low(0) for command.
+
+  pixel-format:
+    type: int
+    default: 0
+    enum:
+      - 0 # RGB565
```

Comment:
Would it be better to use string enums, e.g. "rgb565" and "rgb888", which are much more selfdescriptive?

Reply:
There actually exist definitions in `gc9x91x.h` file. I updated the yaml description to make this more clear.

Reply:
Fixed with latest push